cmake/Findfmt.cmake

```diff
@@ -12,6 +12,23 @@ if(NOT VCPKG_FMT_URL)
     set(VCPKG_FMT_URL "https://github.com/fmtlib/fmt/archive/refs/tags/9.1.0.tar.gz")
 endif()
 
+if(POLICY CMP0135)
+    cmake_policy(SET CMP0135 NEW)
+endif()
+
+set(OLD_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+set(SKIP_WARNINGS OFF)
+if(MSVC AND VCPKG_DEVELOPMENT_WARNINGS AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang") AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "[Cc]lang"))
+    set(SKIP_WARNINGS ON)
+    # fmt\core.h(418): warning C6239: (<non-zero constant> && <expression>) always evaluates to the result of <expression>:  Did you intend to use the bitwise-and (&) operator? If not, consider removing the redundant '<non-zero constant>' and the && operator.
+    string(APPEND CMAKE_CXX_FLAGS " /wd6239")
+    # This one is guarded by an assert
+    # fmt\format-inl.h(327): warning C6385: Reading invalid data from 'pow10_significands'.: Lines: 298, 300, 327
+    string(APPEND CMAKE_CXX_FLAGS " /wd6385")
+    # fmt\os.h(377): warning C6326: Potential comparison of a constant with another constant.
+    string(APPEND CMAKE_CXX_FLAGS " /wd6326")
+endif()
```

Comment:
CMake already has a system of "directory-scoped flags": `add_compile_options()`
```suggestion
if(MSVC AND VCPKG_DEVELOPMENT_WARNINGS AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang") AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "[Cc]lang"))
    # fmt\core.h(418): warning C6239: (<non-zero constant> && <expression>) always evaluates to the result of <expression>:  Did you intend to use the bitwise-and (&) operator? If not, consider removing the redundant '<non-zero constant>' and the && operator.
    add_compile_options(/wd6239)
    # This one is guarded by an assert
    # fmt\format-inl.h(327): warning C6385: Reading invalid data from 'pow10_significands'.: Lines: 298, 300, 327
    add_compile_options(/wd6385)
    # fmt\os.h(377): warning C6326: Potential comparison of a constant with another constant.
    add_compile_options(/wd6326)
endif()
```

Reply:
This isn't "in a directory" though, it needs to edit the cache variables that are passed down through `FetchContent_Declare` -> `ExternalProject_Add`.

Reply:
Interestingly, this created the opposite problem. It did indeed reach down into the `ExternalProject_Add`, but it also applied the effect to vcpkg itself which is undesired:

<img width="1623" alt="image" src="https://user-images.githubusercontent.com/1544943/201203836-36302c26-e64d-4395-af90-1797fd7826fb.png">